### PR TITLE
Trim trailing SP and HTAB for header value on HTTP/1.1

### LINKED ISSUE #77001: SocketsHttpHandler is inconsistent with header value trimming on HTTP/1.1
From https://github.com/dotnet/runtime/pull/74393#discussion_r992509854

The [spec](https://www.rfc-editor.org/rfc/rfc9112#name-field-syntax) defines field lines as
```
  field-line   = field-name ":" OWS field-value OWS
  OWS          = *( SP / HTAB )
```

We currently [trim leading SP and HTAB](https://github.com/dotnet/runtime/blob/d2d5ad3ffb51d345c5c3fa31d7dac118355cfe0d/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L1199-L1204), but only [trailing SP](https://github.com/dotnet/runtime/blob/d2d5ad3ffb51d345c5c3fa31d7dac118355cfe0d/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L1206-L1218).
It should be much more common to encounter an HTAB before the value (following the `name:`), as there usually isn't any whitespace after the value.

Do we care about this inconsistency?
Filing this so we can make a conscious decision we can point to in the future.

### PR DESCRIPTION
**Pull request by Marusyk**
Trim trailing SP and HTAB for header value on HTTP/1.1

Fixes #77001

Please review
Thank you in advance

### CODE REVIEW

**Comment by karelz** [comment]
Thanks @Marusyk for your PR! That was fast :)
Would you mind also adding tests for some interesting cases?

**Comment by Marusyk** [comment]
is the https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs#L612 right place for tests?

**Comment by MihaZupan** [comment]
I would make this a theory and add more cases to it:
https://github.com/dotnet/runtime/blob/960e4d723c27a5407dc691d06e546bc455a9c4a5/src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs#L3683-L3684

File: src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs [hunk 1]
```diff
@@ -19,6 +19,7 @@
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Xml.Linq;
```
Comment (MihaZupan) [inline src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:22]:
```suggestion
```

**Review by MihaZupan — APPROVED** [review]

**Pull request merged**